src/main/webapp/app/exercises/shared/participation/participation.service.ts

```diff
@@ -156,7 +156,7 @@ export class ParticipationService {
     }
 
     public mergeStudentParticipations(participations: StudentParticipation[]): StudentParticipation | undefined {
-        if (participations && participations.length > 0) {
+        if (participations.length) {
```

Comment:
You omitted the check for the participations to be defined. Can you guarantee that the method cannot be called with undefined as argument?

Reply:
The same holds for some other changes below this line

Reply:
The parameter does not allow undefined, otherwise the type must be `StudentParticipation[] | undefined` or the name has to be followed by a question mark

Reply:
It can still happen that some data from the server is retrieved and is handled as it cannot be undefined in the client. So having such (potentially unnecessary) checks is actually a better approach in my opinion. They don't really harm and might prevent unwanted behavior for some unforeseen edge cases

Reply:
I think keeping it is a good idea to cover unforseen edge cases. I suggest to use a more concise syntax:
```suggestion
        if (participations?.length) {
```

Reply:
Done